src_bagl/ui_flow_signMessage712.c

```diff
@@ -1,62 +1,87 @@
-#include "shared_context.h"
-#include "ui_callbacks.h"
+#ifdef HAVE_EIP712_FULL_SUPPORT
 
-void prepare_domain_hash_v0() {
-    snprintf(strings.tmp.tmp, 70, "0x%.*H", 32, tmpCtx.messageSigningContext712.domainHash);
-}
+#include "ui_logic.h"
+#include "shared_context.h"  // strings
+
+enum { UI_712_POS_REVIEW, UI_712_POS_END };
+static uint8_t ui_pos;
 
-void prepare_message_hash_v0() {
-    snprintf(strings.tmp.tmp, 70, "0x%.*H", 32, tmpCtx.messageSigningContext712.messageHash);
+static void dummy_cb(void) {
+    if (!ui_712_next_field()) {
+        if (ui_pos == UI_712_POS_REVIEW) {
+            ux_flow_next();
+            ui_pos = UI_712_POS_END;
+        } else  // UI_712_POS_END
+        {
+            ux_flow_prev();
+            ui_pos = UI_712_POS_REVIEW;
+        }
+    }
 }
 
 // clang-format off
 UX_STEP_NOCB(
-    ux_sign_712_v0_flow_1_step,
+    ux_712_step_review,
     pnn,
     {
-      &C_icon_certificate,
-      "Sign",
+      &C_icon_eye,
+      "Review",
       "typed message",
     });
-UX_STEP_NOCB_INIT(
-    ux_sign_712_v0_flow_2_step,
-    bnnn_paging,
-    prepare_domain_hash_v0(),
-    {
-      .title = "Domain hash",
-      .text = strings.tmp.tmp,
-    });
-UX_STEP_NOCB_INIT(
-    ux_sign_712_v0_flow_3_step,
-    bnnn_paging,
-    prepare_message_hash_v0(),
-    {
-      .title = "Message hash",
+UX_STEP_NOCB(
+   ux_712_step_dynamic,
+   bnnn_paging,
+   {
+      .title = strings.tmp.tmp2,
       .text = strings.tmp.tmp,
-    });
+   }
+);
+UX_STEP_INIT(
+   ux_712_step_dummy,
+   NULL,
+   NULL,
+   {
+    dummy_cb();
```

Comment:
Ok it's actually not a "callback" in the usual UX_STEP sense, but an init, maybe rename to clarify

Reply:
It is used as a callback in this context since this function does not initialize anything, and is never called/referenced explicitly except in the UX_STEP definition and is triggered everytime once this step if reached.